bestbook/api/books.py

```diff
@@ -69,7 +69,16 @@ class Book(core.Base):
 
     @staticmethod
     def clean_olid(olid):
-        """Extract just the olid from some olid-containing string"""
+        """
+        Extract just the olid from some olid-containing string
+        
+        >>> Book.clean_olid(None)
+        ''
+        >>> Book.clean_olid('')
+        ''
+        """
+        if not olid:
+            return ''
         if olid.lower().startswith('ol') and olid.lower()[-1] in ['M', 'W']:
```

Comment:
``` suggestion
        if olid.lower().startswith('ol') and olid[-1] in ['M', 'W']:
```

Reply:
When I look at OLIDs on Open Library, they are uppercase, not lowercase.  So perhaps the logic should be:
```python
if not olid:
    return ''
olid = olid.upper()
[ ... ]
```

Reply:
I think that's great!  Then, all of the persisted values would be formatted consistently.